fire/SFMainMod.F90

```diff
@@ -684,7 +695,7 @@ subroutine  fire_intensity ( currentSite )
           if( hlm_masterproc == itrue ) write(fates_log(),*) 'fire_intensity',currentPatch%fi,W,currentPatch%ROS_front
        endif
        !'decide_fire' subroutine shortened and put in here... 
-       if (currentPatch%FI >= fire_threshold) then  ! 50kW/m is the threshold for a self-sustaining fire
+       if (currentPatch%FI >= fire_threshold .and. currentSite%NF > 0._r8) then !50kW/m threshold for self-sustaining fire
```

Comment:
@jkshuman I see two new lines of code that sum up your changes in this pull request:
1) `currentSite%NF = ED_val_nignitions * years_per_day * CG_strikes` (currently line 680)
2) `if (currentPatch%FI >= fire_threshold .and. currentSite%NF > 0._r8) then` (currently line 698)
The rest is either infrastructure to support these changes or aesthetic improvements.

Assuming that your testing continues to pass, these changes seem fine to me.